api/output_policy_test.go

```diff
@@ -0,0 +1,72 @@
+package api
+
+import (
+	"net/http"
+	"net/url"
+	"testing"
+)
+
+func TestBuildSamplePolicy(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		req      *OutputPolicyError
+		expected string
+		err      error
+	}{
+		{
+			&OutputPolicyError{
+				method: http.MethodGet,
+				path:   "/something",
+			},
+			fmtOutput("/something", []string{"read"}),
+			nil,
+		},
+		{ // test included to clear up some confusion around the sanitize comment
+			&OutputPolicyError{
+				method: http.MethodGet,
+				path:   "http://vault.test/v1/something",
+			},
+			fmtOutput("http://vault.test/v1/something", []string{"read"}),
+			nil,
+		},
+		{ // test that list is properly returned
+			&OutputPolicyError{
+				method: http.MethodGet,
+				path:   "/something",
+				params: url.Values{
+					"list": []string{"true"},
+				},
+			},
+			fmtOutput("/something", []string{"list"}),
+			nil,
+		},
+		{
+			&OutputPolicyError{
+				method: http.MethodGet,
+				path:   "/sys/config/ui/headers/",
+			},
+			fmtOutput("/sys/config/ui/headers/", []string{"read", "sudo"}),
+			nil,
+		},
+		{ // ensure that a formatted path that trims the trailing slash as the code does still works for recognizing a sudo path
+			&OutputPolicyError{
+				method: http.MethodGet,
+				path:   "/sys/config/ui/headers",
+			},
+			fmtOutput("/sys/config/ui/headers", []string{"read", "sudo"}),
+			nil,
+		},
+	}
+
+	for _, tc := range testCases {
```

Comment:
Consider using t.Run() for each test case. You can find examples in similar tests within the repo. This will produce slightly better output when run with `go test -v`.

Reply:
There seems to be a few different testing styles even in this package but I updated to make a new run for each test case here.